frontend/src/useFlow.ts

```diff
@@ -205,35 +228,72 @@ const createSources = async (name: string, originalLanguage: string, translatedL
+    console.log('Created additional source:', additionalSourceEntry);
+  }
+
+  // Create links between all sources and the translated source
+  await postSourceOriginLinks(originalSource.id, additionalSourceIds, translatedSource.id);
+  console.log('Created source origin links for:', {
+    originalSourceId: originalSource.id,
+    additionalSourceIds,
+    translatedSourceId: translatedSource.id,
+  });
+
   return { 
     originalSourceId: originalSource.id,
     translatedSourceId: translatedSource.id,
+    additionalSourceIds,
   };
 };
 
 export function useFlow() {
   const [loadingCount, setLoadingCount] = useState<number>(0);
 
-  const translateSegments = useCallback(async (originalSegments: Segment[], translatedSourceId: number, originalLanguage: string, translatedLanguage: string)
+  const translateSegments = useCallback(async (originalSegments: Segment[], translatedSourceId: number, originalLanguage: string, translatedLanguage: string, additionalSourcesText: ExtractTextResult[] = [])
     : Promise<{translatedSegments: Segment[], translatedSourceId: number}> => {
     setLoadingCount(prev => prev + 1);
     try {
       const translatedSource = await getSource(translatedSourceId);
       const promptKey = "prompt_1"; // Hard-coded default prompt key.
       let promptText = "";
+      const num_additional_sources = additionalSourcesText.length;
       if (translatedSource.dictionary_id) {
-        promptText = await getPrompt({dictionary_id: translatedSource.dictionary_id, dictionary_timestamp: translatedSource.dictionary_timestamp_epoch});
+        promptText = await getPrompt({dictionary_id: translatedSource.dictionary_id, dictionary_timestamp: translatedSource.dictionary_timestamp_epoch, num_additional_sources});
       } else {
-        promptText = await getPrompt({prompt_key: promptKey, original_language: originalLanguage, translated_language: translatedLanguage});
+        promptText = await getPrompt({prompt_key: promptKey, original_language: originalLanguage, translated_language: translatedLanguage, num_additional_sources});
       }
       console.log('Using following prompt: ', promptText);
 
-      const { translated_paragraphs, properties: translationProperties } =
-          await translateParagraphs(originalSegments.map((segment) => segment.text), promptText);
+      // Map additionalSourcesText to format expected by translateParagraphs
+      const additionalSources = additionalSourcesText
+        .filter(source => source.language && source.id) // Only include sources with language and id
+        .map(sourceText => ({
+          text: sourceText.text,
+          language: sourceText.language!,
+          source_id: sourceText.id!,
+        }));
 
+      const { translated_paragraphs, additional_sources_segments, properties: translationProperties } =
+          await translateParagraphs(originalSegments.map((segment) => segment.text), promptText, additionalSources);
```

Comment:
We want to keep translateParagraphs agnostic to our database and structure as it should be also available as external custom API (without our sources_ids...).

So we want to extract the additionalSource from the database and send them as text here.

This should  be as simple as loading segments from several sources